File: placementprepration/tree/replacebst.java
import java.util.*;
class TreeNode{
    int val;
    TreeNode left, right;
    TreeNode(int val){
        this.val=val;
        this.left=null;
        this.right=null;
    }
    public class replacebst{
        public void ino(TreeNode root, List<Integer> vals){
            if(root==null){
                retrun;
            }
            ino(root.left, vals);
            vals.add(root.val);
            ino(root.right,vals);

        }
        public void rep(TreeNode root, List<Integer>vals , int index){
            if(root==null){
                return;
            }
            rep(root.left,vals,index);
            root.val = vals.get(index[0]++);
            rep(root.right,vals,index);
        }
        public void recover(TreeNode root){
            if(root==null){
                return;
            }
            List<Integer>vals=new ArrayList<>();
            ino(root,vals);
            Collections.sort(val);
            int [] index={0};
            rep(root,vals,index);

        }
    }
}

File: placementprepration/tree/replacebst.py
import sys


class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None


def inorder_collect(root, vals):
    if not root:
        return
    inorder_collect(root.left, vals)
    vals.append(root.val)
    inorder_collect(root.right, vals)


def inorder_assign(root, vals_iter):
    if not root:
        return
    inorder_assign(root.left, vals_iter)
    root.val = next(vals_iter)
    inorder_assign(root.right, vals_iter)


def recover_bst(root):
    vals = []
    inorder_collect(root, vals)
    vals.sort()
    inorder_assign(root, iter(vals))


def main():
    # Placeholder: build tree externally, then call recover_bst(root)
    pass


if __name__ == "__main__":
    main()
